.github/workflows/forms-flow-bpm-cd.yml

```diff
@@ -1,92 +1,199 @@
-          # registry: ${{ env.REGISTRY }}
-          username: ${{ secrets.DOCKER_USERNAME }}
-          password: ${{ secrets.DOCKER_ACCESS_TOKEN }}
+          registry: ${{ env.OPENSHIFT_IMAGE_REGISTRY_URL }}/${{ env.BUILD_NAMESPACE }}
+          username: ${{ env.OPENSHIFT_USERNAME }}
+          password: ${{ env.OPENSHIFT_REGISTRY_TOKEN }}
 
       - name: Cache Docker layers
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         with:
           path: /tmp/.buildx-cache
-          key: ${{ runner.os }}-buildx-${{ matrix.name }}-${{ github.sha }}
+          key: ${{ runner.os }}-buildx-${{ env.APP_NAME }}-${{ github.sha }}
           restore-keys: |
-            ${{ runner.os }}-buildx-${{ matrix.name }}
-
+            ${{ runner.os }}-buildx-${{ env.APP_NAME }}
+      
       - name: Build and push Docker image
-        if: ${{ github.ref != 'refs/heads/master' }}
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v3
         with:
-          context: forms-flow-bpm
+          context: ${{ env.APP_NAME }}
           push: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
-      - name: Build and push Docker image
-        if: ${{ github.ref == 'refs/heads/master' }}
-        uses: docker/build-push-action@v2
-        with:
-          context: forms-flow-bpm
-          push: true
-          tags: ${{ steps.meta.outputs.tags }}, formsflow/forms-flow-bpm:latest
-          labels: ${{ steps.meta.outputs.labels }}
+          file: ${{ env.APP_NAME }}/${{ env.DOCKER_FILE_NAME }}
+  
+  # ##################################### Deployment #####################################
+  apply-deploymentConfigs:
+    name: Apply deploymentConfigs
+    if: github.repository == 'bcgov/digital-journeys'
+    environment: ${{ !github.event.inputs.environment && 'dev' ||  github.event.inputs.environment }}
+    runs-on: ubuntu-latest
+    
+    needs:
+      - apply-openshift-buildConfigs
+      - build-and-push-image
+    
+    steps:
+      - name: Checkout Target Branch
+        uses: actions/checkout@v3
+
+      - name: Log in to OpenShift
+        run: |
+          oc login --token=${OPENSHIFT_TOKEN} --server=${OPENSHIFT_SERVER_URL}
+  
+      - name: apply deploymentConfig
+        working-directory: ./deployment/openshift
+        run: |
+          test -n "${NAMESPACE}"
+          test -n "${BUILD_NAMESPACE}"
+          test -n "${BRANCH}"
+          echo "Current ENV is ${ENVIRONMENT}"
+          echo "Current namespace is ${NAMESPACE}"
+          oc -n ${NAMESPACE} project
+          oc -n ${NAMESPACE} process -f bpm_dc.yaml \
+                -p CAMUNDA_ANALYTICS_JDBC_URL=${CAMUNDA_ANALYTICS_JDBC_URL} \
+                -p CAMUNDA_JDBC_USER=${CAMUNDA_JDBC_USER} \
+                -p FORMIO_FILE_URL="${CAMUNDA_FORMIO_FILE_URL}" \
+                -p TOOLS_WORKSPACE=${BUILD_NAMESPACE} \
+                -p SMTP_SERVER="apps.smtp.gov.bc.ca" \
```

Comment:
I suggest moving the hardcoded values to envs at top of the file to ensure one can edit the config in one place.

Reply:
Agree with your point Iman, Move it to top of file along with env section.